doc/source/explanation-flower-next.rst

```diff
@@ -0,0 +1,112 @@
+
+ClientApp
+~~~~~~~~~
+
+Now, onto its counterpart, the ClientApp (|clientapp_link|_). Like the ServerApp, the ClientApp is ephemeral - it is spun up on-demand by the SuperNode to process a message (sent by the ServerApp). When the ClientApp is launched, it receives a message from the SuperNode, executes the instructions in the message, returns results back to the SuperNode, and finally terminates.
+
+.. note::
+    In a future release, we will introduce the concept of multi-app support. This means that multiple ClientApps can be connected to a single SuperNode. This allows multiple users of the same federation to execute different tasks on the same SuperNode, bringing greater freedom for building and using task-specific apps, all while using the same infrastructure.
+
+The reason for employing the mechanisms for launching apps is to enable code reuseability: In previous versions (<1.8), clients are inherently linked to a server for an FL project and additional code needs to be written to move a simulation-based FL setup to a real-world deployment. This can sometimes prohibitively impede the productizing of federated learning workflows. Instead, with the decoupling of the layers in Flower Next, the same ServerApp and ClientApp code can be easily run in a simulation setup or deployment setup *without any code changes*.
+
+
+Information exchange
+--------------------
+
+Messages
+~~~~~~~~
+
+Message (|message_link|_) is an object that Flower uses to carry information between ServerApp and ClientApp. This information can be a model the ServerApp wants to federate, metrics the ClientApp is pushing back to the ServerApp via the SuperLink, and anything in between. The design of Messages and how they are handled by Flower ensures that a Message sent by the ServerApp looks exactly the same when received by the ClientApp (and vice versa). This ensures a more unified and smoother developer experience.
+
+Context
+~~~~~~~
+
+Context (|context_link|_) is another useful object that we introduced in Flower Next. For every execution of `ServerApp`, i.e. a run, it carries the record and messages that is unique to the run. Each time a SuperNode runs a ClientApp within the run, the same Context object is exposed to the ClientApp, allowing the ClientApp to persist throughout the duration of the run. Context can be used as a temporary buffer for executing additional tasks, like `mods`. 
```

Comment:
I would reference (https://flower.ai/docs/framework/how-to-use-built-in-mods.html) after mentioning mods.